airflow-core/src/airflow/utils/helpers.py

```diff
@@ -204,7 +204,7 @@ def build_airflow_dagrun_url(dag_id: str, run_id: str) -> str:
     http://localhost:8080/dags/hi/runs/manual__2025-02-23T18:27:39.051358+00:00_RZa1at4Q
     """
     baseurl = conf.get("api", "base_url", fallback="/")
-    return urljoin(baseurl, f"dags/{dag_id}/runs/{run_id}")
+    return urljoin(f"{baseurl}/", f"dags/{dag_id}/runs/{run_id}")
```

Comment:
`baseurl` might contain a trailing slash. I guess it's better to add it when not present or use `rstrip` and then always add slash.

```
task-sdk/src/airflow/sdk/execution_time/execute_workload.py:    default_execution_api_server = f"{base_url.rstrip('/')}/execution/"
airflow-core/src/airflow/executors/local_executor.py:    default_execution_api_server = f"{base_url.rstrip('/')}/execution/"
```

Related

https://github.com/apache/airflow/pull/54248/files#r2262038890
https://github.com/apache/airflow/pull/53884